lib/cartopy/io/img_tiles.py

```diff
@@ -322,26 +322,31 @@ def _image_url(self, tile):
 
 class Stamen(GoogleWTS):
     """
-    Retrieves tiles from maps.stamen.com. Styles include
+    Retrieves tiles from stadiamaps.com. Styles include
+    ``stamen_terrain-background``, ``stamen_terrain``,
+    ``stamen_toner``, and ``stamen_watercolor``. Previously,
+    these were available as
     ``terrain-background``, ``terrain``, ``toner`` and ``watercolor``.
 
     For a full reference on the styles available please see
-    http://maps.stamen.com. Of particular note are the sub-styles
-    that are made available (e.g. ``terrain`` and ``terrain-background``).
-    To determine the name of the particular [sub-]style you want,
-    follow the link on http://maps.stamen.com to your desired style and
-    observe the style name in the URL. Your style name will be in the
-    form of: ``http://maps.stamen.com/{STYLE_NAME}/#9/37/-122``.
+    https://docs.stadiamaps.com/themes/. Of particular note are the sub-styles
+    that are made available (e.g. ``stamen_terrain`` and
+    ``stamen_terrain_background``).
 
     Except otherwise noted, the Stamen map tile sets are copyright Stamen
-    Design, under a Creative Commons Attribution (CC BY 3.0) license.
+    Design, under a Creative Commons Attribution (CC BY 4.0) license.
 
     Please see the attribution notice at http://maps.stamen.com on how to
     attribute this imagery.
+    For Toner and Terrain:
+    Map tiles by Stamen Design, under CC BY 4.0. Data by OpenStreetMap, under ODbL.
+    For Watercolor:
+    Map tiles by Stamen Design, under CC BY 4.0. Data by OpenStreetMap, under CC BY SA.
```

Comment:
The updated Stamen styles have some updated attribution requirements, which you can find here: https://docs.stadiamaps.com/attribution/. We've also shortened the text for each required attribution, since there are more of them.

Attributions needed for all Stadia Maps styles
* Stadia Maps
* OpenMapTiles
* OpenStreetMap

Additionally, Stamen styles require a credit to Stamen. The page linked above has recommended text.

Reply:
Thanks for chiming in here! I just pinged you on #2269 which I opened with the new style class for this. I put in the updated attribution text, let me know if you want more in the docstring or I missed something.